### PR TITLE
Must-gather, Store PIDs of all the subprocesses

### PR DESCRIPTION
**Pull request by OdedViner**

### CODE REVIEW

File: must-gather/collection-scripts/gather [hunk 1]
```diff
@@ -17,32 +20,40 @@ for ns in $(oc get dataprotectionapplications.oadp.openshift.io --all-namespaces
...
 fi
 
 # Gather version information
 echo "Gathering version information for top-level debug file"
 /usr/bin/gather_versions "/must-gather/clusters/${clusterID}/oadp-debug-info" ${skip_tls} &
+pids+=($!)
+
+# Check if PID array has any values, if so, wait for them to finish
+if [ ${#pids[@]} -ne 0 ]; then
+    echo "Waiting on subprocesses to finish execution."
```
Comment (kaovilai) [inline must-gather/collection-scripts/gather:53]: Is it worth since we're tracking PIDs to store what we're waiting on? ie. waiting on X step

Comment (OdedViner) [reply must-gather/collection-scripts/gather:53]: I think we need to wait for specific pids finish...

Comment (kaovilai) [reply must-gather/collection-scripts/gather:53]: I meant to say is it also viable to echo the PID and process names so we know what we're stuck on

**Comment by kaovilai** [comment]
/cc @mrnold 
any comments?

**Comment by kaovilai** [comment]
/cc @mrnold 
any comments?

**Comment by mrnold** [comment]
I am curious what prompted this? Like is there a bug link or something so I can see what this change fixes?

I really like Tiger's idea of showing the name of the current process being waited on, so we can see if a job is stuck.

**Comment by OdedViner** [comment]
> I am curious what prompted this? Like is there a bug link or something so I can see what this change fixes?
> 
> I really like Tiger's idea of showing the name of the current process being waited on, so we can see if a job is stuck.

Hi @mrnold, Hi opened some PR to must-gather in OCP so I wanted that all MG code will work with the same method  https://github.com/openshift/must-gather/commit/4c685e04ae7dd236782a1b09912bd2f95c3c5690

**Comment by mrnold** [comment]
> Hi @mrnold, Hi opened some PR to must-gather in OCP so I wanted that all MG code will work with the same method [openshift/must-gather@4c685e0](https://github.com/openshift/must-gather/commit/4c685e04ae7dd236782a1b09912bd2f95c3c5690)

I see, this should be fine then.

**Review by kaovilai — APPROVED** [review]

**Pull request merged**